here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/RequestSender.java

```diff
@@ -114,12 +114,49 @@ public boolean hasKeyProps(KeyProperties thatKeyProps) {
   /**
    * Set of properties that are just enough to construct the sender
    * and distinguish unambiguously between objects
-   * in terms of their effective configuration
+   * in terms of their effective configuration.
+   * Objects of this class are compared based on their contents, not on object reference.
    */
-  public record KeyProperties(
-      @NotNull String name,
-      @NotNull String hostUrl,
-      @NotNull Map<String, String> defaultHeaders,
-      long connectionTimeoutSec,
-      long socketTimeoutSec) {}
+  public static class KeyProperties {
```

Comment:
Why convert this record to a class?
If it is about being compared based on the contents, java records already override `equals` method
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Record.html#equals(java.lang.Object)

Reply:
It's because the timeout parameters should be modified based on supplied `SessionOptions` in `HttpStorage.newReadSession()`, and a record is immutable.

Reply:
As the `RequestSender` is shared between multiple sessions it cannot be mutable.

If the `RequestSender` configuration is only established at the session creation stage, not at the `HttpStorage` stage, my suggestion is to remove `requestSender` field from `HttpStorage` and create new request sender each `newReadSession`.
Of course using `RequestSenderCache.getInstance().getSenderWith(...)`

Reply:
thanks, will adjust so, including reverting KeyProperties back to record